### ISSUE TITLE
[CP] Ignore invalid description in vswhere.exe JSON output

### ISSUE THREAD

**Issue by loic-sharma** [flutter/flutter#106920]
### issue_link

https://github.com/flutter/flutter/issues/106601

### Commit Hash

dcbdff0

### Target

stable

### Impacted Users

Windows developers using specific locales

### Impact Description

This impacts development: `flutter doctor`, `flutter run`, and `flutter build` do not work on Windows for certain locales.

Only Chinese and Japanese locales are known to be affected by this issue. However, it is likely that other locales are also affected by this.

### Workaround

Yes, workarounds include:

1. Downgrade Visual Studio to an unaffected version
2. Switch your locale to English
3. Patch your Visual Studio installation and replace vswhere.exe with an unreleased build: https://github.com/microsoft/vswhere/releases

### Risk

low

### Test Coverage

yes

### Validation Steps

This fix is covered by unit tests. It can also be verified manually by running `flutter doctor` on Windows.

**Comment by itsjustkevin** [comment]
@christopherfujino could you take a look at this cp?

**Comment by christopherfujino** [comment]
> @christopherfujino could you take a look at this cp?

I can't speak to the product need for this, but it seems like an annoyance, and not worth cherrypicking. Most likely the user will know they have Visual Studio installed.

@loic-sharma does this bug actually block building for windows?

**Comment by loic-sharma** [comment]
Yes this blocks building for Windows as that depends on [Visual Studio's CMake](https://github.com/flutter/flutter/blob/d6f090d952a9e82f9456d0fa311eed815116a2d9/packages/flutter_tools/lib/src/windows/visual_studio.dart#L155-L172), whose [installation path is detected using vswhere.exe](https://github.com/flutter/flutter/blob/d6f090d952a9e82f9456d0fa311eed815116a2d9/packages/flutter_tools/lib/src/windows/visual_studio.dart#L295-L306).

**Comment by christopherfujino** [comment]
> Yup this blocks building for Windows as that depends on [Visual Studio's CMake](https://github.com/flutter/flutter/blob/d6f090d952a9e82f9456d0fa311eed815116a2d9/packages/flutter_tools/lib/src/windows/visual_studio.dart#L155-L172), whose [installation path is detected using vswhere.exe](https://github.com/flutter/flutter/blob/d6f090d952a9e82f9456d0fa311eed815116a2d9/packages/flutter_tools/lib/src/windows/visual_studio.dart#L295-L306).

Oops, you state that in the description :P

**Comment by christopherfujino** [comment]
This LGTM

**Comment by loic-sharma** [comment]
> Oops, you state that in the description :P

No worries, the description wasn't super detailed so asking for more information is perfectly reasonable! 😄
